controlplane/kubeadm/internal/workload_cluster.go

```diff
@@ -173,20 +176,35 @@ func (w *Workload) getConfigMap(ctx context.Context, configMap ctrlclient.Object
 }
 
 // UpdateImageRepositoryInKubeadmConfigMap updates the image repository in the kubeadm config map.
-func (w *Workload) UpdateImageRepositoryInKubeadmConfigMap(ctx context.Context, imageRepository string, version semver.Version) error {
-	return w.updateClusterConfiguration(ctx, func(c *bootstrapv1.ClusterConfiguration) {
+func (w *Workload) UpdateImageRepositoryInKubeadmConfigMap(imageRepository string) func(*bootstrapv1.ClusterConfiguration, *[]string) {
+	return func(c *bootstrapv1.ClusterConfiguration, updatedKeys *[]string) {
 		if imageRepository == "" {
 			return
 		}
+
 		c.ImageRepository = imageRepository
-	}, version)
+		*updatedKeys = append(*updatedKeys, "imageRepository")
```

Comment:
I will drop updatedKeys for a couple of reasons.
- It might be misleading because ultimately it is UpdateClusterConfiguration makes diff and decides if to update or not 
- It might be misleading because UpdateClusterConfiguration always updates the entire config map
- It could be confusing, because updated doesn't always means changed

Last but not least, it clean up the func signature.

If we want some more visibility, we can log (with V 3 or 4) whenever a value is actually changed, but this can be also a follow up